README.md

```diff
@@ -144,33 +144,112 @@ $ make -C build.assets build-binaries
 
 ### Local Build
 
-To perform a build on your host, ensure you have installed Go. In order to
-include the Rust-powered features like Desktop Access, you'll
-also need `cargo` and `rustc`. The current versions of these tools can be found
-in `build.assets/Makefile`.
+**Important:**
 
-```bash
-# get the source & build:
+* The Go compiler is somewhat sensitive to the amount of memory: you will need
+  **at least** 1GB of virtual memory to compile Teleport. A 512MB instance
+  without swap will **not** work.
+* This will build the latest version of Teleport, **regardless** of whether it
+  is stable. If you want to build the latest stable release, run `git checkout`
+  to the corresponding tag (for example, run `git checkout v8.0.0`) **before**
```

Comment:
Might want to add `git submodule --update --recursive` as well to make sure web assets are correct.

Reply:
This is entirely specific to Teleport employees though, right? Do we want to include this into public repo README?

Reply:
The webassets submodule is public, so I would say yes.

Reply:
But not the `e` submodule, right?

Reply:
Right, but `e` is not required to do an OSS teleport build.

Reply:
Sure, however `git submodule --update --recursive` will fail for anyone but Teleport Employees?

Reply:
Anyways, I added a note about it.